pkg/generate/app/imagelookup.go

```diff
@@ -154,7 +154,7 @@ func (r DockerRegistryResolver) Resolve(value string) (*ComponentMatch, error) {
 
 	from := ref.Registry
 	if len(ref.Registry) == 0 {
-		ref.Registry = "Docker Hub"
+		from = "Docker Hub"
```

Comment:
Besides not wanting to hard code anything specific to the Docker Hub in here, this isn't going to work because `ref.Registry` might be populated. If it is populated, it could point to the Hub (docker.io), it could point to an external registry (myregistry.com), or it could point to our integrated registry (172.30.x.x:5000 or a route for it). I think you should be creating the input ImageStream if the ref has no registry or if its registry doesn't point to the OpenShift integrated registry. @smarterclayton thoughts?